planner_cspace/src/planner_3d.cpp

```diff
@@ -1502,16 +1493,88 @@ class Planner3dNode
   bool makePlan(const geometry_msgs::Pose& gs, const geometry_msgs::Pose& ge,
                 nav_msgs::Path& path, bool hyst)
   {
-    Astar::Vec s, e;
-    grid_metric_converter::metric2Grid(
-        map_info_, s[0], s[1], s[2],
-        gs.position.x, gs.position.y, tf2::getYaw(gs.orientation));
-    s.cycleUnsigned(map_info_.angle);
+    Astar::Vec e;
     grid_metric_converter::metric2Grid(
         map_info_, e[0], e[1], e[2],
         ge.position.x, ge.position.y, tf2::getYaw(ge.orientation));
     e.cycleUnsigned(map_info_.angle);
 
+    Astar::Vecf sf;
+    grid_metric_converter::metric2Grid(
+        map_info_, sf[0], sf[1], sf[2],
+        gs.position.x, gs.position.y, tf2::getYaw(gs.orientation));
+    Astar::Vec s(static_cast<int>(floor(sf[0])), static_cast<int>(floor(sf[1])), lroundf(sf[2]));
+    s.cycleUnsigned(map_info_.angle);
+
+    std::vector<Astar::VecWithCost> starts;
+    if (antialias_start_)
+    {
+      const int x_cand[] =
+          {
+            0, ((sf[0] - s[0]) < 0.5 ? -1 : 1)
+          };
+      const int y_cand[] =
+          {
+            0, ((sf[1] - s[1]) < 0.5 ? -1 : 1)
+          };
+      for (const int x : x_cand)
+      {
+        for (const int y : y_cand)
+        {
+          const Astar::Vec p = s + Astar::Vec(x, y, 0);
+          const Astar::Vecf subpx = sf - Astar::Vecf(p[0] + 0.5f, p[1] + 0.5f, 0.0f);
+          if (subpx.sqlen() > 1.0)
+            continue;
+          if (cm_[p] > 99)
+            continue;
+
+          starts.push_back(Astar::VecWithCost(p));
+        }
+      }
+    }
+    else
+    {
+      if (cm_[s] < 100)
+      {
+        starts.push_back(Astar::VecWithCost(s));
+      }
+    }
+    for (Astar::VecWithCost& s : starts)
+    {
+      const Astar::Vecf diff =
+          Astar::Vecf(s.v_[0] + 0.5f, s.v_[1] + 0.5f, 0.0f) - sf;
+      s.c_ = hypotf(diff[0] * ec_rough_[0], diff[1] * ec_rough_[1]);
+      s.c_ += cm_[s.v_] * cc_.weight_costmap_ / 100.0;
+
+      // Check if arrived to the goal
+      Astar::Vec remain = s.v_ - e;
+      remain.cycle(map_info_.angle);
+      if (remain.sqlen() <= goal_tolerance_lin_ * goal_tolerance_lin_ &&
+          abs(remain[2]) <= goal_tolerance_ang_)
```

Comment:
You should use std::abs() or fabs(), because the argument of abs() (without std::) is int and remain[2] is implicitly cast here.

Reply:
`remain[2]` and `goal_tolerance_ang_` are in grid coordinate integer.
`std::abs` would be better anyway.

Reply:
`fabs -> std::abs` kind of things: https://github.com/at-wat/neonavigation/issues/354